### ISSUE TITLE
Manual zoom and "fit to height" mode video width limit

### ISSUE THREAD

**Issue by ShadowMaster85** [tamius-han/ultrawidify#141]
I am using the new version 5.0.0 of the extension, the zoom keys Z and U now are working, but the zoom stopped working when the with of the video reach the with of the screen even if there is other black space at the top and at the bottom of the video, the "fit to height" mode instead limit the width of the video to the width of the screen wich altrer the aspect ratio, i want to zoom the video to the full height of the screen but without limit the width of the video to the width of the screen because it alter the aspect ratio, i have tryed an old nightly version of the extension wich at least let me do a single zoom before limit the width of the video as described before, what can i do? there is a solution or a workaround?

    Browser/browser version: Opera 74.0.3911.218 Official Portable
    Operating system: Windows 10 Home 1709
    Ultrawidify version: 5.0.0 from Chrome store

**Comment by tamius-han** [comment]
TL;DR Chrome version of the addon limits zoom in order to dodge some Chrome bugs. Use Firefox if you want more zoom.

> stopped working when the with of the video reach the with of the screen even if there is other black space at the top and at the bottom of the video

This is "by design" (actually, there should be 5-10 black pixels at the edges).

Back in January, Chrome had a bug with hardware acceleration. If all of the following were true:

* the video got both taller and wider than the monitor
* while in fullscreen mode
* when hardware acceleration was enabled
* if using nVidia GPU

it would just get stretched and not zoomed in (basically, you'd get the equivalent of 'basic stretch' if the video got zoomed in too much).

Since that didn't happen if the video never got wider than `width_of_montior * 0.997` or something like that, the video width is limited to `width_of_montior * 0.997` pixels.

### The rumor has it

It is said that this issue is fixed for Chrome (and also Edge, Edge also had it!) since early February or something, but I got messaged about that problem as late as mid-march (and another extension similar to mine, except without autodetection, still has the 'video gets stretched' issue that got comments as recently as this week) ... so I don't think I'll be removing the limit yet. 

### Worth noting (bonus rant)

Even though Opera is basically a Chrome reskin since like 2012, I don't think I've seen that bug happen in Opera when I tested it out. Unfortunately though, there is no Opera branch of Ultrawidify. 

Not for the lack of trying. I've actually tried submitting the extension to the opera addons, but it appears that Opera Addons moderation team consists solely of [Gaben](https://knowyourmeme.com/memes/half-life-3-confirmed) and [George R.R. Martin](https://en.wikipedia.org/wiki/The_Winds_of_Winter).

The extension still hasn't passed _or failed_ the moderator review over a year later (first submitted: 2020-03-14), so I kinda stopped bothering with maintaining the separate Opera version for the _one_ Opera Next user who doesn't use Chrome Web Store version of the addon.

**Comment by ShadowMaster85** [comment]
I have an Intel integrated GPU so this chrome/chromium bug should not be present in my case right? anyhow there is a mode to manually disable that zoom limit, like a value to change or to add in the advanced settings?, so i can test if in my case the zoom limit removal can help!
Thanks!

**Comment by tamius-han** [comment]
No, for the time being there's not (since this was a very hastily added fix).

**Comment by tamius-han** [comment]
As of 5.0.1, it is possible to disable the max zoom limitation. Since the issue is fixed in chrome, and since it has been fixed for a while, I've decided to remove the limitation by default, even.

_However,_ it turns out that this issue has popped up in Opera, which means that people who use Opera on nVidia hardware are not going to be happy (you probably won't be affected since you say you use integrated GPU), but fuck it. Chrome Web Store is for Google Chrome extensions, not my fault Opera doesn't have a functional addon store.
